packages/react/menu/src/Menu.tsx

```diff
@@ -435,16 +437,62 @@ const MenuContentImpl = React.forwardRef((props, forwardedRef) => {
     ...contentProps
   } = props;
   const context = useMenuContext(CONTENT_NAME);
-  const typeaheadProps = useMenuTypeahead();
   const { getItems } = useCollection();
   const [currentItemId, setCurrentItemId] = React.useState<string | null>(null);
   const contentRef = React.useRef<HTMLDivElement>(null);
   const composedRefs = useComposedRefs(forwardedRef, contentRef, context.onContentChange);
   const isPointerDownOutsideRef = React.useRef(false);
+  const timerRef = React.useRef(0);
+  const searchRef = React.useRef('');
+  const [isSearching, setIsSearching] = React.useState(false);
 
   const PortalWrapper = portalled ? Portal : React.Fragment;
   const ScrollLockWrapper = disableOutsideScroll ? RemoveScroll : React.Fragment;
 
+  // Reset `searchRef` 1 second after it was last updated
+  const setSearch = React.useCallback((search: string) => {
```

Comment:
I know you copied this from before but do you mind if we rename this to `handleSearchChange` or something. I thought that when we were calling `setSearch` on line 469 that we were calling a state setter and I couldn't see that on existed, I got lost haha.